docs/specs/workflow.md

```diff
@@ -318,3 +318,99 @@ And then the frontend will receive a response to telling it to redirect.
 
 Finally the frontend will be visited again with `https://myapp.com/oauth-redirect`.
 The frontend can include necessary information in the redirect URI to make the resumption possible.
+
+### Intent for login id migration (Experimental)
+
+The Intent for login id migration allows the developer to create a workflow for migration.
+
+[![](https://mermaid.ink/img/pako:eNqdUl1rwkAQ_CvLPdv2PQ9Cq4JS_EArfWhK2SYbc2ju0ru9ioj_vZvGaCVgSwMHuZ2ZnZu926vEpqQi5ekjkEmor3HlsIgNyFeiY53oEg3DfeB8RejaSC94tsVy1EaG1q4BPfTJWLiDZ3rPpVLzGtVNt9u0jqDnCJlAGybDYy0nkd3WunW2sdta15BF17SIYE4cnDkxYdTvSBPPKIlkc8Vx5mxClALnZyPYas7hafo4mLQ8q0iiQu9rArCtY35qhHO6qnRi96wcxYWE4eVVp5JM8-72mG5RUgJoUoFQXCowQbbuAs-cLWA2nE4Gb5Pl-GEwvzA5hzmO4X8urah_GVJV8SL2v1_Otfv4SfahoAunil0t1VEFuQJ1Ku91_11VwisoVpH8ppRh2HCsYnMQquS0i51JVCSjp44KZSpBj89bRRluPB2-AAamCc8?type=png)](https://mermaid.live/edit#pako:eNqdUl1rwkAQ_CvLPdv2PQ9Cq4JS_EArfWhK2SYbc2ju0ru9ioj_vZvGaCVgSwMHuZ2ZnZu926vEpqQi5ekjkEmor3HlsIgNyFeiY53oEg3DfeB8RejaSC94tsVy1EaG1q4BPfTJWLiDZ3rPpVLzGtVNt9u0jqDnCJlAGybDYy0nkd3WunW2sdta15BF17SIYE4cnDkxYdTvSBPPKIlkc8Vx5mxClALnZyPYas7hafo4mLQ8q0iiQu9rArCtY35qhHO6qnRi96wcxYWE4eVVp5JM8-72mG5RUgJoUoFQXCowQbbuAs-cLWA2nE4Gb5Pl-GEwvzA5hzmO4X8urah_GVJV8SL2v1_Otfv4SfahoAunil0t1VEFuQJ1Ku91_11VwisoVpH8ppRh2HCsYnMQquS0i51JVCSjp44KZSpBj89bRRluPB2-AAamCc8)
+
+#### Configuration in `authgear.yaml`
+
+```yaml
+migration_workflow:
+  url: authgeardeno:///deno/migration.ts
+  timeout: 10
+```
+
+The developer can pass the request body to the workflow. Authgear will send the request body to the hook, and the hook should return the identities' and authenticators' MigrateSpec.
```

Comment:
For simplicity, I think we just need to support `migration_token`, which is always a string.

Reply:
Looks good to me. And for the hook request, we will put the migration_token as the request body without further serialization. Thought?

Reply:
You mean `{ "migration_token": "TOKEN" }`?